backend/Origam.Architect.Server/Services/DeploymentScriptRunnerService.cs

```diff
@@ -0,0 +1,145 @@
+#region license
+/*
+Copyright 2005 - 2025 Advantage Solutions, s. r. o.
+
+This file is part of ORIGAM (http://www.origam.org).
+
+ORIGAM is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+ORIGAM is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with ORIGAM. If not, see <http://www.gnu.org/licenses/>.
+*/
+#endregion
+
+using MoreLinq;
+using Origam.Architect.Server.Enums;
+using Origam.DA.Service;
+using Origam.Schema.DeploymentModel;
+using Origam.Workbench.Services;
+using Origam.Workbench.Services.CoreServices;
+
+namespace Origam.Architect.Server.Services;
+
+public class DeploymentScriptRunnerService(ILogger<DeploymentScriptRunnerService> logger)
+{
+    private static readonly log4net.ILog log = log4net.LogManager.GetLogger(
+        System.Reflection.MethodBase.GetCurrentMethod()?.DeclaringType
+    );
+
+    public void RunDeploymentScript(AbstractUpdateScriptActivity script)
+    {
+        var transactionId = Guid.NewGuid().ToString();
+
+        try
+        {
+            logger.LogInformation("Executing deployment script: {ScriptName}", script.Name);
+            ExecuteActivity(script, transactionId);
+            ResourceMonitor.Commit(transactionId);
+            logger.LogInformation(
+                "Deployment script executed successfully: {ScriptName}",
+                script.Name
+            );
+        }
+        catch (Exception ex)
+        {
+            ResourceMonitor.Rollback(transactionId);
+            logger.LogError(ex, "Error executing deployment script: {ScriptName}", script.Name);
+            throw new Exception($"Error executing deployment script '{script.Name}'", ex);
+        }
+    }
+
+    private void ExecuteActivity(AbstractUpdateScriptActivity activity, string transactionId)
+    {
+        if (log.IsInfoEnabled)
```

Comment:
This can be replaced with an ILogger call:

        if(logger.IsEnabled(LogLevel.Info))